### PR TITLE
Coadd 2 objects

### PR DESCRIPTION
**Pull request by profxj**
Modifies code to better coadd one pair of objects.

Additional enhancements.

New docs.

### CODE REVIEW

**Comment by profxj** [comment]
Am finding other issues to fix.  
Closing for now.

**Comment by profxj** [comment]
Reopening with a substantial refactor

**Comment by profxj** [comment]
@tiffanyhsyu , please have a look now.

I've moved away (but not entirely) from using XSpectrum1D
as the masking was making my head hurt.

And I caught at least one significant bug.

File: pypit/arcoadd.py [hunk 1]
```diff
@@ -241,24 +261,55 @@ def grow_mask(initial_mask, n_grow=1):
...
+    allok = okm & okf
+    # Ratio
+    med_flux = fluxes[iref,allok] / fluxes[ispec,allok]
+    # Clip
+    mn_scale, med_scale, std_scale = sigma_clipped_stats(med_flux, sigma=nsig, iters=niter, **kwargs)
+    # Return
+    return med_scale
+
+
+def median_flux(spec, smask, nsig=3., niter=5, **kwargs):
```
Comment (tiffanyhsyu) [inline pypit/arcoadd.py]: Can we remove this, since it's not used anymore?

**Comment by tiffanyhsyu** [comment]
Sorry this review took so long!

I was stuck on the new unpack_spec and more generally, the lines that do .filled(0.). I was worried that any flux values that were set to 0.0 were included when the individual exposures were coadded. So, for example, if we were coadding 3 spectra, at an index i with flux values, 100., 90., and something filled to 0., the 0. would be included in the coadded spectrum and bias the resultant coadded flux value.

I think I've convinced myself that any flux values filled to 0. are actually masked out though (when defining rmask), so this shouldn't be a problem.

Anyway, this looks good to me.

File: pypit/arcoadd.py (hunk 1 above)
Comment (profxj) [inline pypit/arcoadd.py]: I have now commented it out.

**Comment by profxj** [comment]
Merging.

**Pull request merged**